### PR TITLE
add support for tuple of types on KORNIA_CHECK_TYPE + bug fix

### PR DESCRIPTION
**Pull request by cjpurackal**
#### Changes

Fixes # (issue)

#### Type of change

- [ ] 📚  Documentation Update
- [ ] 🧪 Tests Cases
- [x] 🐞 Bug fix (non-breaking change which fixes an issue)
- [x] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

#### Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

File: kornia/core/check.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 """The testing package contains testing-specific utilities."""
 from __future__ import annotations
 
-from typing import Any, Sequence, TypeVar, cast
+from typing import Any, Sequence, Tuple, TypeVar, cast
```
Comment (johnnv1) [inline kornia/core/check.py:4]:
```suggestion
from typing import Any, Sequence, TypeVar, cast
```

File: kornia/core/check.py [hunk 2]
```diff
@@ -102,7 +102,7 @@ def KORNIA_UNWRAP(maybe_obj: object, typ: Any) -> Any:
 T = TypeVar('T', bound=type)
 
 
-def KORNIA_CHECK_TYPE(x: object, typ: T, msg: str | None = None) -> TypeGuard[T]:
+def KORNIA_CHECK_TYPE(x: object, typ: T | tuple[T, ...], msg: str | None = None) -> TypeGuard[T]:
```
Comment (johnnv1) [inline kornia/core/check.py:105]: It probably won't work with for the typeguard (mypy), but I don't know how it could be done either 😄

Comment (cjpurackal) [reply kornia/core/check.py:105]: Is it not passing mypy?

Comment (johnnv1) [reply kornia/core/check.py:105]: example:

```python
from typing import TypeVar, TypeGuard, Type


def myf(a: int | str) -> int:
    return 1


T = TypeVar('T')


def KORNIA_CHECK_TYPE(x: object, typ: Type[T] | tuple[Type[T], ...]) -> TypeGuard[T]:
    if not isinstance(x, typ):
        raise
    return True


var_A = True
if KORNIA_CHECK_TYPE(var_A, (int, str)):
    print(myf(var_A))  # L19: not guard properly the typing
if KORNIA_CHECK_TYPE(var_A, int):
    print(myf(var_A))  # L21: works fine
if isinstance(var_A, (int, str)):
    print(myf(var_A))  # L24 works fine


var_B = 1
if KORNIA_CHECK_TYPE(var_B, (int, str)):
    print(myf(var_B))  # L28: not guard properly the typing
if isinstance(var_B, (int, str)):
    print(myf(var_B))  # L30: works fine
```

mypy output:
```bash
$ mypy t.py 
t.py:19: error: Argument 1 to "myf" has incompatible type "object"; expected "Union[int, str]"  [arg-type]
t.py:28: error: Argument 1 to "myf" has incompatible type "object"; expected "Union[int, str]"  [arg-type]
```


The problem here is something like this: the `typ` tuple is a tuple of `type_A`, `type_B`, ..., `type_N` -- so how we can properly type this for the type guard can guarantee that the `x` object is one of these `N` types

Comment (johnnv1) [reply kornia/core/check.py:105]: i think we can leave it as it is, just add a TODO comment. I am not sure if there is a solution for this typing, anyway some time would have to be put on top of this to find a solution

**Review by johnnv1 — APPROVED** [review]

**Pull request merged**